Nginx custom ports

ameijer-corsha opened the pull request:
### Proposed changes
Addresses #580 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto master
- [X] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

pleshakov commented:
@ameijer-corsha thanks for the PR! we will review it shortly

pleshakov commented on deployments/helm-chart/README.md:
```diff
@@ -94,6 +94,7 @@ Parameter | Description | Default
 `controller.service.externalIPs` | The list of external IPs for the Ingress controller service. | []
 `controller.service.loadBalancerSourceRanges` | The IP ranges (CIDR) that are allowed to access the load balancer. Requires `controller.service.type` set to `LoadBalancer`. The cloud provider must support this feature. | []
 `controller.service.name` | The name of the service. | Autogenerated
+`controller.service.customPorts` | A list of external ports to expose to the Ingress controller. Follows conventional Kubernetes yaml service port syntax | {}
```
```suggestion
`controller.service.customPorts` | A list of custom ports to expose through the Ingress controller service. Follows the conventional Kubernetes yaml syntax for service ports. | []
```

pleshakov commented on deployments/helm-chart/values.yaml:
```diff
@@ -157,6 +157,9 @@ controller:
       ## The HTTPS port on the POD where the Ingress controller service is running.
       targetPort: 443
 
+    ## yaml definition of custom ports to expose in the service
```
the doc should match the doc string in the table in the README.

pleshakov commented on deployments/helm-chart/values.yaml:
```diff
@@ -157,6 +157,9 @@ controller:
       targetPort: 443
 
+    ## yaml definition of custom ports to expose in the service
+    customPorts: {}
```
customPorts is an array - `[]`

ameijer-corsha commented:
@pleshakov Thank you for the review! I believe I have addressed all your concerns. I rendered this a few times, all looks good. Could you review again when you have the chance? much appreciated...

pleshakov commented:
@ameijer-corsha thx! looks good!
 could you possible squash your commits? we will merge once it is done

pleshakov commented:
@ameijer-corsha could you possibly change the commit message to "Support custom ports in helm chart"? In the message body, you can add "Closes #580", which will close the issue once the commit is merged. Please see our git style guide https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md#git-style-guide

We're trying to keep the commit history consistent https://github.com/nginxinc/kubernetes-ingress/commits/master Thanks for understanding!

pleshakov commented:
@ameijer-corsha thanks!

ameijer-corsha commented:
@pleshakov is there a roadmap for when this feature will make it into the stable chart? we are trying to plan our prod deployment

pleshakov commented:
@ameijer-corsha 
Next major release 1.6.0 is planned for the week of Dec 16th. It will include all the features that were added to the master branch.
